prov/shm/src/smr_ep.c

```diff
@@ -132,12 +132,28 @@ int smr_setopt(fid_t fid, int level, int optname,
 	struct smr_ep *smr_ep =
 		container_of(fid, struct smr_ep, util_ep.ep_fid);
 
-	if ((level != FI_OPT_ENDPOINT) || (optname != FI_OPT_MIN_MULTI_RECV))
+	if ((level != FI_OPT_ENDPOINT))
 		return -FI_ENOPROTOOPT;
 
-	smr_get_smr_srx(smr_ep)->min_multi_recv_size = *(size_t *)optval;
+	if (optname == FI_OPT_MIN_MULTI_RECV) {
```

Comment:
is the multi_recv stuff here unrelated, or is it because this PR changes the way we enable/disable shm because of hmem changes?